modules/console/pages/ui/programmable-push-filters.adoc

```diff
@@ -2,58 +2,151 @@
 :page-aliases: console:features/programmable-push-filters.adoc, reference:console/programmable-push-filters.adoc
 // Do not put page aliases in the single-sourced content
 // tag::single-source[]
-:description: Learn how to filter Kafka records in {ui} based on your provided JavaScript code.
+:description: Learn how to filter Kafka records using custom JavaScript code within {ui}.
 
-You can use push-down filters in {ui} to search for specific records within a Kafka topic.
+You can use push-down filters in {ui} to search through large Kafka topics that may contain millions of records. Filters are JavaScript functions executed on the backend, evaluating each record individually. Your function must return a boolean:
 
-Push-down filters are TypeScript/JavaScript function bodies that you define in {ui} and that are executed on the backend for
-every individual record in a topic. The code must return a boolean. If your code returns `true`, the backend sends the record to the results in the frontend.
-Otherwise the record is skipped and {ui} continues to consume records until either the selected number
-of maximum search results or the end of the topic has been reached.
+* `true`: record is included in the frontend results.
+* `false`: record is skipped.
 
-On a topic's *Messages* page, click *Add filter* > *JavaScript Filter*.
+Multiple filters combine logically with `AND` conditions.
 
-{ui} can inject the following properties into your function, which you can use in your filter code:
+== Add a JavaScript filter
 
-* `partitionId` - The record's partition ID
-* `offset` - The record's offset within its partition
-* `key` - The record's key in its decoded form
-* `value` - The record's value in its decoded form
-* `headers` - The record's header value in its decoded form
+To add a JavaScript filter:
 
-NOTE: Keys, values, and headers are passed into your JavaScript code in their decoded form. The
-deserialization logic (for example, decode an Avro serialized byte array to a JSON object) is applied first, before injecting it into
-the JavaScript function. If your record is presented as a JSON object in the UI, you can also access it
-like a JavaScript object in your filter code.
+. Navigate to the topic's *Messages* page.
+. Click *Add filter > JavaScript Filter*.
+. Define your JavaScript filtering logic in the provided input area.
 
-Suppose you have a series of Avro, JSON, or Protobuf encoded record values that deserialize to JSON objects like this:
+image::ROOT:console:js-filter.png[alt="JavaScript filter in {ui}"]
 
-[,json]
+== Resource usage and performance
+JavaScript filters are executed on the backend, consuming CPU and network resources. The performance of your filter depends on the complexity of your JavaScript code and the volume of data being processed.
+Typical performance is *15,000-20,000 filtered records/second per CPU core*.
```

Comment:
Where are we getting these numbers from? I would refrain from using this.
I think general note that it may impact performance is fine, but stating any specifics could be tricky as it depends on a lot of factors.
Also we may want to specify that the evaluation is done within _Console_ backend and not _within_ Redpanda broker just so it is clear.

Reply:
Those numbers were in the original doc. I'll remove them since they are estimates as you said.

Reply:
Oh ok sorry I was reading the rendered output more than the diff. I think maybe not having specifics is still a good idea.